File: example/led_blinky.py
import time

LED = "/sys/class/leds/uvled/"

def main():
    with open(LED + "trigger", "w") as f:
        f.write("none\n")
        f.flush()
        
    with open(LED + "brightness", "w") as f:
        while True:
            f.write("1\n")
            f.flush()
            time.sleep(1)
            f.write("0\n")
            f.flush()
            time.sleep(1)
            
if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        with open(LED + "brightness", "w") as f:
            f.write("0\n")
            f.flush()
        with open(LED + "trigger", "w") as f:
            f.write("heartbeat\n")
            f.flush()

File: example/led_blinky.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <signal.h>
#include <fcntl.h>
#include <string.h> // Added header

#define LED "/sys/class/leds/uvled/"

int keep_running = 1;

void int_handler(int dummy) {
    keep_running = 0;
}

void write_to_file(const char *path, const char *value) {
    int fd = open(path, O_WRONLY);
    if (fd < 0) {
        perror("open");
        exit(EXIT_FAILURE);
    }
    if (write(fd, value, strlen(value)) < 0) {
        perror("write");
        close(fd);
        exit(EXIT_FAILURE);
    }
    close(fd);
}

int main() {
    signal(SIGINT, int_handler);

    write_to_file(LED "trigger", "none\n");

    while (keep_running) {
        write_to_file(LED "brightness", "1\n");
        sleep(1);
        write_to_file(LED "brightness", "0\n");
        sleep(1);
    }

    write_to_file(LED "brightness", "0\n");
    write_to_file(LED "trigger", "heartbeat\n");

    return 0;
}
